oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/UpdateUtils.java

```diff
@@ -150,6 +150,18 @@ public static boolean checkConditions(@NotNull Document doc,
                 } else if (c.type == Condition.Type.NOTEQUALS && equal) {
                     return false;
                 }
+            } else if (c.type == Condition.Type.LESSTHAN) {
+                if (r != null) {
+                    if (value instanceof Map) {
+                        value = ((Map) value).get(r);
+                    } else {
+                        value = null;
+                    }
+                }
+                if (value instanceof Comparable && c.value instanceof Comparable) {
+                    int lessThan = ((Comparable) value).compareTo(c.value);
+                    return lessThan == -1;
+                }
```

Comment:
I think this should not silently ignore cases when value is not Comparable, but this could be enforced earlier.

Reply:
I have removed all the logic related with "lessThan", as now I'm using UpdateOp.max() as you proposed.